storage/grpc_client.go

```diff
@@ -1718,25 +1718,26 @@ func (c *grpcStorageClient) OpenWriter(params *openWriterParams, opts ...storage
 	}
 	s.retry.maxRetryDuration = retryDeadline
 
+	// Set Flush func for use by exported Writer.Flush.
+	var gw *gRPCWriter
+	setFlush(func() (int64, error) {
+		return gw.flush()
+	})
+
 	// This function reads the data sent to the pipe and sends sets of messages
 	// on the gRPC client-stream as the buffer is filled.
 	go func() {
 		err := func() error {
-			var gw *gRPCWriter
-			gw, err := newGRPCWriter(c, s, params, pr, pw, params.setPipeWriter)
+			var err error
+			gw, err = newGRPCWriter(c, s, params, pr, pw, params.setPipeWriter)
```

Comment:
I think we should move this out of the go func as well

Reply:
Addressed this while keeping the error handling the same...